src/adagrams.js

```diff
@@ -1,15 +1,176 @@
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+// helper function
+const createLetterPool = (letterPool) => {
+  let letterPoolList = [];
+  for (const [letter, frequency] of Object.entries(letterPool)) {
+    for (let i = 0; i < frequency; i++) {
+      letterPoolList.push(letter);
+    }
+  }
+  return letterPoolList;
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let letterPoolList = createLetterPool(letterPool);
+  let currentHand = [];
+  for (let i = 0; i < 10; i++) {
+    let poolIndex = Math.floor(Math.random() * (letterPoolList.length - 1) + 1);
+    let drawLetter = letterPoolList[poolIndex];
+    currentHand.push(drawLetter);
+    const index = letterPoolList.indexOf(drawLetter);
+    letterPoolList.splice(index, 1);
+  }
+
+  return currentHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  let handCopy = Object.assign(lettersInHand);
```

Comment:
you could also use the spread operator here which is what you will commonly see esxpecially in javascript. Here is a link to a resource that shows many ways to make a copy and 8 shows how to make a deep copy: https://www.freecodecamp.org/news/how-to-clone-an-array-in-javascript-1d3183468f6a/

using a spread operator
```javascript
let lettersInHandCopy = [...lettersInHand];
```